src/Message/Uri.php

```diff
@@ -173,7 +173,7 @@ public function withHost($host)
             return $this;
         }
 
-        if (\preg_match('#[\s_%+]#', $host) || ($host !== '' && \parse_url('http://' . $host, \PHP_URL_HOST) !== $host)) {
+        if (\preg_match('#[\s%+]#', $host) || ($host !== '' && \parse_url('http://' . $host, \PHP_URL_HOST) !== $host)) {
```

Comment:
Could you please add a unit test covering this so we don't regress in the future?

Reply:
I added a corresponding test case in the valid URI provider

Reply:
Awesome thanks 👍